backend/main.py

```
@@ -830,6 +841,62 @@ async def delete_memory(project_id: str, memory_id: str):
+
+        final_text = await project_detail_service.ingest_project_detail(
+            project_id=project_id,
+            raw_text=raw_text,
+            mode=(request.mode or "merge")
+        )
+        return {"status": "saved", "chars": len(final_text), "mode": (request.mode or "merge").lower()}
+    except HTTPException:
+        raise
+    except Exception as e:
+        logger.error(f"Error ingesting project detail: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to ingest project detail: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)